src/main/java/org/bytedeco/javacpp/tools/Parser.java

```diff
@@ -96,7 +97,17 @@ public Parser(Logger logger, Properties properties, String encoding, String line
     InfoMap leafInfoMap = null;
     TokenIndexer tokens = null;
     String lineSeparator = null;
-    Set<String> upcasts = new HashSet<>();
+    Set<String> upcasts = new HashSet<>(); // Java names of base classes needing upcast
+    Map<String, Set<Type>> downcasts = new HashMap<>(); // Keys: classes which child classes needs downcast constructor. Values: classes to downcast from.
+    HashSet<String> polymorphicClasses = new HashSet<>(); // Contains Java names
```

Comment:
Declare that as `Set` instead of `HashSet` for consistency, and while we're at it, use Javadoc-style comments

Reply:
We can put the comments on a single line still, let's do that